Lake Water Quality conversion fails with assertion error in xarray Zarr backend

Conversion using the attached file produces "AssertionError: We should never get here. Function logic must be wrong." (see traceback)

[config.yaml.zip](https://github.com/bcdev/nc2zarr/files/6011221/config.yaml.zip)
[traceback.txt](https://github.com/bcdev/nc2zarr/files/6011223/traceback.txt)

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,13 +2,26 @@
 
 ### Version 0.2.0 (in development)
 
-* Fixed some issues with Zarr output chunk sizes given by process 
-  parameter `rechunk`: (#12)
-  - Fix: New chunking can now overlap multiple chunks in input (Dask chunks).
+* Fixed some issues with Zarr (re-)chunking given by process parameter
+  `process/rechunk`: 
+  - Fix: New chunking can now overlap multiple chunks in input 
+    (Dask chunks). (#12)
   - Introduced new chunk size value `"input"` which forces using chunk sizes 
-    from input.
+    from input. (#12)
+  - Fixed problems that occurred when using the `process/rechunk` 
+    configuration
+    + if a dimension was not configured at all;
+    + if the dataset contained 0-dimension variables, which is 
+      typical for variables used to describe the spatial reference system, 
+      e.g. `crs`.
+      
+    In these cases the error message from Zarr was
+    `AssertionError: We should never get here."`. (#14)
+
 * Introduced new input parameter `prefetch_chunks`. (#10) 
+
 * Add an AppVeyor CI configuration file.
+
 * Add the `--sort-by` command-line option.
 
 ### Version 0.1.0 (08.01.2021)

--- a/nc2zarr/processor.py
+++ b/nc2zarr/processor.py
@@ -81,9 +81,10 @@ class DatasetProcessor:
                 elif not isinstance(dim_chunk_size, int):
                     raise ValueError(f'invalid chunk size: {dim_chunk_size}')
                 chunks.append(dim_chunk_size)
-            chunks = tuple(chunks)
-            ds_rechunked[k] = v.chunk(chunks)
-            output_encoding[var_name] = dict(chunks=chunks)
+            if chunks:
+                chunks = tuple(chunks)
+                ds_rechunked[k] = v.chunk(chunks)
+                output_encoding[var_name] = dict(chunks=chunks)
         return ds_rechunked, output_encoding
 
     @classmethod

